src/trails.erl

```diff
@@ -151,18 +153,61 @@ trails(Handler) ->
   trails([Handler], []).
 
 %% @doc Store the given list of trails.
--spec store(trails()) -> ok.
+-spec store(Trails::trails() |
+              [{HostMatch::route_match(), Trails::trails()}]) -> ok.
 store(Trails) ->
   _ = application:ensure_all_started(trails),
+  store('_', Trails).
+
+-spec store(Server::ranch:ref(),
+            Trails::trails() |
+              [{HostMatch::route_match(), Trails::trails()}]) -> ok.
+store(_Server, []) ->
+  ok;
+store(Server, [{HostMatch, Trails} | Hosts]) ->
+  NormalizedPaths = normalize_store_input(Trails),
+  do_store(Server, HostMatch, NormalizedPaths),
+  store(Server, Hosts);
+store(Server, Trails) when is_list(Trails) ->
```

Comment:
You don't need to check that `Trails` is a list. That's already defined in the function spec (therefore dialyzer will catch any possible type error) and we should not program defensively.